src/main/webapp/app/pages/curation/nestLevelSummary/MutationLevelSummary.tsx

```diff
@@ -3,25 +3,36 @@ import { IRootStore } from 'app/stores';
 import { MutationLevelSummary } from 'app/stores/firebase/firebase.gene.store';
 import _ from 'lodash';
 import { observer } from 'mobx-react';
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import NestLevelSummary from './NestLevelSummary';
+import { onValue, ref } from 'firebase/database';
+import { getMutationStats } from 'app/shared/util/firebase/firebase-utils';
 
 export interface MutationLevelSummaryProps extends StoreProps {
-  mutationUuid: string;
+  mutationPath: string;
   hideOncogenicity?: boolean;
 }
 
-const MutationLevelSummary = ({ mutationSummaryStats, mutationUuid, hideOncogenicity = false }: MutationLevelSummaryProps) => {
-  const mutationLevelSummary = mutationSummaryStats[mutationUuid];
-  return <NestLevelSummary summaryStats={mutationLevelSummary} hideOncogenicity={hideOncogenicity} />;
+const MutationLevelSummary = ({ mutationPath, firebaseDb, hideOncogenicity = false }: MutationLevelSummaryProps) => {
+  const [mutationStats, setMutationStats] = useState(undefined);
+
+  useEffect(() => {
+    const callbacks = [];
+    callbacks.push(
+      onValue(ref(firebaseDb, mutationPath), snapshot => {
+        const calcMutationStats = getMutationStats(snapshot.val());
```

Comment:
Thoughts on creating a delayed calculation, like debounce here?

Reply:
Same can be applied to other **Summary

Reply:
I agree, will save some work and not recalculate every keystroke.